packages/doenetml-worker-rust/lib-doenetml-core/src/core/math_via_wasm.rs

```diff
@@ -0,0 +1,327 @@
+///
+/// assert_eq!(
+///     math_to_latex(&expr_with_blanks, ToLatexParams::default()).unwrap(),
+///     "x + \u{FF3F}"
+/// );
+///
+/// let hide_blanks = ToLatexParams {
+///     show_blanks: false,
+///     ..Default::default()
+/// };
+/// assert_eq!(
+///     math_to_latex(&expr_with_blanks, hide_blanks).unwrap(),
+///     "x + "
+/// );
+/// ````
+#[cfg(all(not(feature = "testing"), feature = "web"))]
+pub fn math_to_latex<MathObj: AsRef<str>>(
+    math_object: MathObj,
+    params: ToLatexParams,
+) -> Result<String, anyhow::Error> {
+    let result: JsString = toLatex(
+        JsString::from(math_object.as_ref()),
+        params.pad_to_decimals.map(Number::from),
+        params.pad_to_digits.map(Number::from),
+        Boolean::from(params.show_blanks),
+    )
+    .map_err(|e| anyhow!("{:?}", e))?;
+    Ok(result.into())
+}
+#[cfg(any(feature = "testing", not(feature = "web")))]
+pub fn math_to_latex<MathObj: AsRef<str>>(
+    _math_object: MathObj,
+    _params: ToLatexParams,
+) -> Result<String, anyhow::Error> {
+    Err(anyhow!(
+        "math_to_latex is only available when compiled with the `web` feature".to_string(),
+    ))
+}
+
+/// Create a new mathematical expression formed by substituting variables with new expressions
+///
+/// Parameters:
+/// - `math_object`: a serialized `math-expressions` object
+/// - `substitutions`: a `HashMap` mapping variables to new expressions
+///
+/// Example:
+///
+/// ```
+/// let expr1 = parse_text_into_math("x+y", true, &["f"]).unwrap();
+///
+/// let substitutions =
+///   HashMap::from([("y".to_string(), MathOrPrimitive::VariableName("z".to_string()))]);
+/// let expr2 = substitute_into_math(expr1, &substitutions).unwrap();
+///
+/// assert_eq!(
+///    math_to_latex(&expr2, ToLatexParams::default()).unwrap(),
+///    "x + z"
+/// );
+/// ```
+#[cfg(all(not(feature = "testing"), feature = "web"))]
+pub fn substitute_into_math<MathObj: AsRef<str>>(
+    math_object: MathObj,
+    substitutions: &HashMap<String, MathOrPrimitive>,
+) -> Result<String, anyhow::Error> {
+    let substitutions = serde_wasm_bindgen::to_value(&substitutions).unwrap();
+
+    let result: JsString = substituteIntoMath(JsString::from(math_object.as_ref()), substitutions)
+        .map_err(|e| anyhow!("{:?}", e))?;
+    Ok(result.into())
+}
+#[cfg(any(feature = "testing", not(feature = "web")))]
+pub fn substitute_into_math<MathObj: AsRef<str>>(
+    _math_object: MathObj,
+    _substitutions: &HashMap<String, MathOrPrimitive>,
+) -> Result<String, anyhow::Error> {
+    Err(anyhow!(
+        "substitute_into_math is only available when compiled with the `web` feature".to_string()
+    ))
+}
+
+/// Return a normalize mathematical expression from `math_object` as specified in `params`.
+///
+/// Example:
+///
+/// ```
+/// let expr = parse_text_into_math("(x+x+1)(2y+1-y)", true, &["f"]).unwrap();
+///
+/// let lp = ToLatexParams::default();
```

Comment:
Why do all your examples store `lp` to a variable instead of just putting `ToLatexParams::default()` into where `lp` is used? In any case, this should be defined much closer to its use.

Reply:
It was there to make the examples easier to read! But, since apparently this failed, I got rid of the `lp`s.